### PR TITLE
Add Mayhem

### PR DESCRIPTION
**Pull request by nottirb**
- Added a Dockerfile for fuzz testing
- Added Mayhemfile(s) for fuzz testing
- Added a `mayhem.yml` file for automatically running fuzz tests with Mayhem using the `mcode` action

### CODE REVIEW

File: src/sqlparser/fuzz/fuzz_targets/fuzz_parse_sql.rs [hunk 1]
```diff
@@ -13,14 +13,12 @@
 // limitations under the License.
```
Comment (xansec) [inline src/sqlparser/fuzz/fuzz_targets/fuzz_parse_sql.rs:13]: Can you comment on why the changes to this fuzz target were made?

**Review by xansec — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: src/sqlparser/fuzz/fuzz_targets/fuzz_parse_sql.rs (hunk 1 above)
Comment (nottirb) [reply src/sqlparser/fuzz/fuzz_targets/fuzz_parse_sql.rs:13]: ``GenericDialect`` was removed from the code base. The function ``parse_sql`` only takes in 1 argument instead of 2 now. The changes I made to the fuzz target is just an update to make it work with the existing codebase.

Forgot to mention that in the initial PR post. Sorry about that!

**Review by xansec — APPROVED** [review]
LGTM!

**Comment by xansec** [comment]
@nottirb This action is failing:
https://github.com/mayhemheroes/risingwave/runs/7532414112?check_suite_focus=true

According to the output, I believe the simple fix is to make sure that the package stage of the Dockerfile installs openssl and pkg-config. Could you please open another PR resolving this? I'll get it reviewed once I see it.

**Comment by nottirb** [comment]
> @nottirb This action is failing: https://github.com/mayhemheroes/risingwave/runs/7532414112?check_suite_focus=true
> 
> According to the output, I believe the simple fix is to make sure that the package stage of the Dockerfile installs openssl and pkg-config. Could you please open another PR resolving this? I'll get it reviewed once I see it.

I'll look into this, and open another PR when I get it fixed. Thanks for bringing it to my attention!
